pkg/webhooks/ingresscontroller/ingresscontroller.go

```diff
@@ -149,12 +176,151 @@ func (wh *IngressControllerWebhook) authorized(request admissionctl.Request) adm
 		}
 	}
 
-	ret = admissionctl.Allowed("IngressController operation is allowed")
+	/* TODO:
+	 * - HypershiftEnabled is currently set to false/disabled.
+	 *   If HCP is to be enabled for allowed source ranges, should this part of a 2nd ingress validator to
+	 *   allow separation of validations between cluster install types? ...or if there's a reliable run time method available
+	 *   to this validator to determine classic vs hcp they can remain in the single webhook(?)
+	 * - Classic vs HCP could likely share some of the network funcions, but will need slightly
+	 *   different logic as well as permissions fetching the network config info from different
+	 *   source  (configmap) locations and config formats(?).
+	 */
+	// Only check for machine cidr in allowed ranges if creating or updating resource...
+	reqOp := request.AdmissionRequest.Operation
+	if reqOp == admissionv1.Create || reqOp == admissionv1.Update {
+		//TODO: Will these need to iterate over more than just the default IngressController config?
+		if ic.ObjectMeta.Name == "default" && ic.ObjectMeta.Namespace == "openshift-ingress-operator" {
+			ret := wh.checkAllowsMachineCIDR(ic.Spec.EndpointPublishingStrategy.LoadBalancer.AllowedSourceRanges)
+			ret.UID = request.AdmissionRequest.UID
+			if !ret.Allowed {
+				log.Info("Error checking minimum AllowedSourceRange", "err", ret.AdmissionResponse.String())
+			}
+			return ret
+		}
+	}
+	log.Info("############# DEBUG LOG: IngressController operation is allowed ###########")
+	ret = admissionctl.Allowed("IngressController operation is allowed, machineCIDR n/a")
 	ret.UID = request.AdmissionRequest.UID
 
 	return ret
 }
 
+func (wh *IngressControllerWebhook) getMachineCIDR() (net.IP, *net.IPNet, error) {
+	if wh.machineCIDRIP == nil || wh.machineCIDRNet == nil {
```

Comment:
Rather than lazy-loading the CIDR on first request, would it make sense to just initialize these values in the `NewWebhook()` function? 

The benefits I see with that approach (correct me if I'm wrong on any of these points, though):
- We can omit the `kubeClient` field from the webhook, since it looks like the only time it's used is when we call `getClusterConfig`
- we'll know at hook startup if anything goes wrong w/ the retrieval, rather than at first request (which may make it slightly easier to identify issues here)
- we don't sacrifice any performance on the first request

Reply:
Thanks. I've made a change making the call during the init/NewWebHook() func exit with error if we fail to load and parse the machineCIDR. This should now only fetch and parse the config once at init time. Testing this now.